qiskit_experiments/tomography/tomography_experiment.py

```diff
@@ -0,0 +1,307 @@
+
+        Args:
+            circuit: the quantum process circuit. If not a quantum circuit
+                it must be a class that can be appended to a quantum circuit.
+            measurement_basis: Tomography basis for measurements.
+            measurement_qubits: Optional, the qubits to be measured. These should refer
+                to the logical qubits in the state circuit.
+            preparation_basis: Tomography basis for measurements.
+            preparation_qubits: Optional, the qubits to be prepared. These should refer
+                to the logical qubits in the process circuit.
+            basis_indices: Optional, the basis elements to be measured. If None
+                All basis elements will be measured.
+            qubits: Optional, the physical qubits for the initial state circuit.
+        """
+        # Initialize BaseExperiment
+        if qubits is None:
+            qubits = circuit.num_qubits
+        super().__init__(qubits)
+
+        # Get the target tomography circuit
+        if isinstance(circuit, QuantumCircuit):
+            target_circuit = circuit
+        else:
+            # Convert input to a circuit
+            num_qubits = circuit.num_qubits
+            target_circuit = QuantumCircuit(num_qubits)
+            target_circuit.append(circuit, range(num_qubits))
+        self._circuit = target_circuit
+
+        # Measurement basis and qubits
+        self._meas_circ_basis = measurement_basis
+        if measurement_qubits:
+            self._meas_qubits = tuple(measurement_qubits)
+        else:
+            self._meas_qubits = None
+
+        # Preparation basis and qubits
+        self._prep_circ_basis = preparation_basis
+        if preparation_qubits:
+            self._prep_qubits = tuple(preparation_qubits)
+        else:
+            self._prep_qubits = None
+
+        # Configure experiment options
+        if basis_indices:
+            self.set_experiment_options(basis_indices=basis_indices)
+
+        # Compute target state
+        self._target_state = None
+        if self._prep_circ_basis:
+            self._target_state = self._target_quantum_channel(
+                self._circuit,
+                measurement_qubits=self._meas_qubits,
+                preparation_qubits=self._prep_qubits,
+            )
+        else:
+            self._target_state = self._target_quantum_state(
+                self._circuit, measurement_qubits=self._meas_qubits
+            )
+
+        # Configure analysis basis options
+        analysis_options = {}
+        if measurement_basis:
+            analysis_options["measurement_basis"] = measurement_basis
+        if preparation_basis:
+            analysis_options["preparation_basis"] = preparation_basis
+        self.set_analysis_options(**analysis_options)
+
+    def _metadata(self):
+        metadata = super()._metadata()
+        if self._target_state:
+            metadata["target_state"] = copy.copy(self._target_state)
```

Comment:
perhaps `_expected_outcome` for state and channel? or can we assign different key for channel? saying _state_ for channel seems to be bit misleading.